appsmithorg/appsmith
# fix: New API button env condition

## Description
> User was not able to configure Staging to connect to Dropbox with OAuth2.0.
> Server was not able to access current editing environment to save and authorise the DS.
> We have update the /code endpoint to include current editing environment. This will help server to access the required environment.
>
#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
>
## Testing
>
#### How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Also list any relevant details for your test configuration.
> Delete anything that is not relevant
>
>
#### Test Plan
1. Created a new DS and checked New API functionality in CE
2. In EE, Keep env the dropdown at Staging, create a new OAuth DS and configure it for Prod. From review page, click on New API and ensure it takes user to to the API screen and ensure that the API string field is not having any reference to the DS as it was not configured for Staging
3. Change the dropdown to Production and ensure that the API string is updated properly to the Prod Database configuration.
4. Keep the env dropdown at Production, create a new OAuth DS and configure it for Staging. From review page [env dropdown is Prod], click on New API and ensure it takes user to to the API screen and ensure that the API string field is not having any reference to the DS as it was not configured for Production
5. Change the dropdown to Staging and ensure that the API string is updated properly to the Staging Database configuration.
6. Try the above on a new Workspace with a new application
7. Try the above on an imported application with the existing OAuth DS
8. Try the above on an imported application with a new OAuth DS
>
>
#### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)
>
>
>
## Checklist:
#### Dev activity


#### QA activity:
